packages/validator/src/util/externalSignerClient.ts

```diff
@@ -160,17 +162,68 @@ export async function externalSignerUpCheck(remoteUrl: string): Promise<boolean>
     headers: {"Content-Type": "application/json"},
   });
 
-  const data = await handlerExternalSignerResponse<{status: string}>(res);
+  const data = await handlerExternalSignerUpCheckResponse(res);
   return data.status === "OK";
 }
 
-async function handlerExternalSignerResponse<T>(res: Response): Promise<T> {
+async function handlerExternalSignerGetKeysResponse<T>(res: Response, logger?: Logger): Promise<string[]> {
   if (!res.ok) {
     const errBody = await res.text();
+    logger?.debug("External signer GetKeys error", {body: errBody});
     throw Error(`${errBody}`);
   }
 
-  return JSON.parse(await res.text()) as T;
+  const resBody = await res.text();
+  const resContentType = res.headers.get("Content-Type");
+
+  logger?.debug("External signer GetKeys response", {contentType: resContentType, body: resBody});
+
+  if(resContentType?.startsWith("application/json")) {
+    return JSON.parse(resBody) as string[];
+  }
+  else {
+    return resBody.split(",");
```

Comment:
According to the remote signer spec, this is not a valid response, get keys only support json responses
https://github.com/ethereum/remote-signing-api/blob/7879ed3284bf88acb1149d30a70fbf8b554c183b/signing/paths/public_keys.yaml#L11

I am not sure if this is a shortcoming in the spec or just Diva not following it. At least for signature requests it supports both content types

Reply:
I knew this wasn't in the spec, but I was wondering if I should throw an exception or try to "play nice" and parse a text/plain response. Diva implements the JSON response as in the spec, so it is not nessery and I removed it. Now it throws an error if the content type is invalid.

Reply:
It's strange that this endpoint does not support text/plain, but imo it's bad in the first place that we allow text/plain responses, I am guessing this was required for backward compatibility with web3signer or other existing remote signers.